### PR TITLE
Add option for explicit chmod with atomic file write.

### LINKED ISSUE #80: Directories created via pebble with requested permissions 0777 get created with 0775 instead
I've found that creating directories via the make-dirs action (POST @ /v1/files) may not respect the requested permissions.  Specifically, during testing I found that if I tried to create a directory with 0777 permissions, it got created with 0775 permissions instead.

### PR DESCRIPTION
**Pull request by rwcarlsen**
And use this explicit chmod flag for file writes coming from the files api.

Fixes #80

### CODE REVIEW

**Review by benhoyt — CHANGES_REQUESTED** [review]
Thanks! Just a couple of minor things: one bug-fix (missing `err :=`) and request for a files-API specific test.

File: internal/osutil/io.go [hunk 1]
```diff
@@ -95,6 +98,13 @@ func NewAtomicFile(filename string, perm os.FileMode, flags AtomicWriteFlags, ui
 		return nil, err
 	}
 
+	if flags&AtomicWriteChmod != 0 {
+		fd.Chmod(perm)
```
[the remark this answers is not in the document]
Comment (rwcarlsen) [reply internal/osutil/io.go]: Oops - fixed.

File: internal/osutil/io_test.go [hunk 2]
```diff
@@ -104,6 +105,20 @@ func (ts *AtomicWriteTestSuite) TestAtomicWriteFileAbsoluteSymlinks(c *C) {
 	c.Assert(p, testutil.FileEquals, "hi")
 }
 
+func (ts *AtomicWriteTestSuite) TestAtomicWriteFileChmod(c *C) {
+	tmpdir := c.MkDir()
+	oldmask := syscall.Umask(222)
+	defer syscall.Umask(oldmask)
+
+	p := filepath.Join(tmpdir, "foo")
```
[the remark this answers is not in the document]
Comment (rwcarlsen) [reply internal/osutil/io_test.go]: done.

File: internal/osutil/io_test.go [hunk 3]
```diff
@@ -104,6 +105,20 @@ func (ts *AtomicWriteTestSuite) TestAtomicWriteFileAbsoluteSymlinks(c *C) {
...
+	err := osutil.AtomicWriteFile(p, []byte(""), 0777, osutil.AtomicWriteChmod)
```
[the remark this answers is not in the document]
Comment (rwcarlsen) [reply internal/osutil/io_test.go]: done.

**Review by benhoyt — APPROVED** [review]
Great, thanks!

**Comment by rwcarlsen** [comment]
Is this waiting on anything in particular?

**Comment by benhoyt** [comment]
A Gustavo review. Though this is really a minor bug fix, not an API change, so I'd say we can probably push it through.

**Comment by benhoyt** [comment]
Merged, thanks @rwcarlsen!

**Pull request merged**